main/lsp/LSPTypechecker.cc

```diff
@@ -94,10 +94,15 @@ void LSPTypechecker::initialize(TaskQueue &queue, std::unique_ptr<core::GlobalSt
 
     // Unblock the indexer now that its state is fully initialized.
     {
+        // We copy our file table over to the indexer to ensure that we start with a consistent view of the workspace:
+        // the indexer uses presence in the file table to determine if a file is new or not.
+        auto files = gs->getFiles().subspan(1);
```

Comment:
I wonder if we should push this `.subspan(1)` into `getFiles` so that callers don't have to think about the file at index 0 that doesn't exist.

Reply:
I think the problem is that we'd have to audit all uses to make sure we weren't also looping with an index that we construct a `FileRef` with. When we finally get to c++20, we could make a range that returned a pair of a `FileRef` and a `std::shared_ptr<const File>`...